core/types/gen_header_rlp.go

```diff
@@ -0,0 +1,56 @@
+// Code generated by rlpgen. DO NOT EDIT.
+
+//go:build !norlpgen
+// +build !norlpgen
+
+package types
+
+import "github.com/ethereum/go-ethereum/rlp"
+import "io"
+
+func (obj *Header) EncodeRLP(_w io.Writer) error {
+	w := rlp.NewEncoderBuffer(_w)
+	_tmp0 := w.List()
+	w.WriteBytes(obj.ParentHash[:])
+	w.WriteBytes(obj.UncleHash[:])
+	w.WriteBytes(obj.Coinbase[:])
+	w.WriteBytes(obj.Root[:])
+	w.WriteBytes(obj.TxHash[:])
+	w.WriteBytes(obj.ReceiptHash[:])
+	w.WriteBytes(obj.Bloom[:])
+	if obj.Difficulty == nil {
+		w.Write(rlp.EmptyString)
+	} else {
+		if obj.Difficulty.Sign() == -1 {
+			return rlp.ErrNegativeBigInt
+		}
+		w.WriteBigInt(obj.Difficulty)
+	}
+	if obj.Number == nil {
+		w.Write(rlp.EmptyString)
+	} else {
+		if obj.Number.Sign() == -1 {
+			return rlp.ErrNegativeBigInt
+		}
+		w.WriteBigInt(obj.Number)
+	}
+	w.WriteUint64(obj.GasLimit)
+	w.WriteUint64(obj.GasUsed)
+	w.WriteUint64(obj.Time)
+	w.WriteBytes(obj.Extra)
+	w.WriteBytes(obj.MixDigest[:])
+	w.WriteBytes(obj.Nonce[:])
+	_tmp1 := obj.BaseFee != nil
+	if _tmp1 {
+		if obj.BaseFee == nil {
+			w.Write(rlp.EmptyString)
```

Comment:
This looks like it's unreachable code. Known quirk? Not that it matters a whole lot, if it makes the encoder logic simpler.